src/checkoutservice/main.go

```diff
@@ -184,6 +198,17 @@ func (cs *checkoutService) PlaceOrder(ctx context.Context, req *pb.PlaceOrderReq
 		Items:              prep.orderItems,
 	}
 
+	shippingCostFloat, _ := strconv.ParseFloat(fmt.Sprintf("%d.%02d", prep.shippingCostLocalized.GetUnits(), prep.shippingCostLocalized.GetNanos()/10000000), 64)
+	totalPriceFloat, _ := strconv.ParseFloat(fmt.Sprintf("%d.%02d", total.GetUnits(), total.GetNanos()/10000000), 64)
+
+	span.SetAttributes(
+		attribute.String("app.order.id", orderID.String()),
+		attribute.String("app.order.tracking.id", shippingTrackingID),
+		attribute.Float64("app.order.shipping.cost", shippingCostFloat),
+		attribute.Float64("app.order.total.cost", totalPriceFloat),
```

Comment:
Nit: For all the attributes you've already used above in span`.AddEvent` calls that are also here, maybe you could save them to a local var to avoid the repetition.